pkg/collector/corechecks/system/gpu/nvidia/jetson/jetson.go

```diff
@@ -119,55 +119,33 @@ func (c *JetsonCheck) processTegraStatsOutput(tegraStatsOuptut string) error {
 
 // Run executes the check
 func (c *JetsonCheck) Run() error {
-	// Kill tegrastats if it runs for twice as long as the interval we specified, to avoid blocking
-	// the check forever
-	ctx, cancel := context.WithTimeout(context.Background(), 2*tegraStatsInterval*time.Millisecond)
-	defer cancel()
-	cmd := exec.CommandContext(ctx, c.tegraStatsPath, c.commandOpts...)
-
-	// Parse the standard output for the stats
-	stdout, err := cmd.StdoutPipe()
-	if err != nil {
-		return err
+	tegraStatsCmd := fmt.Sprintf("%s %s", c.tegraStatsPath, strings.Join(c.commandOpts, " "))
+	cmdStr := fmt.Sprintf("(%s) & pid=$!; (sleep %d && kill -9 $pid)", tegraStatsCmd, int((2 * tegraStatsInterval).Seconds()))
+	var cmd *exec.Cmd
+	if c.useSudo {
+		// -n, non-interactive mode, no prompts are used
+		cmd = exec.Command("sudo", "-n", "sh", "-c", cmdStr)
+	} else {
+		cmd = exec.Command("sh", "-c", cmdStr)
 	}
-	go func() {
-		in := bufio.NewScanner(stdout)
-		if in.Scan() {
-			// We only need to read one line
-			line := in.Text()
-			log.Debugf("tegrastats: %s", line)
-			if err = c.processTegraStatsOutput(line); err != nil {
-				log.Error(err)
-			}
-		} else {
-			log.Warnf("tegrastats did not produce any output")
-		}
-		// Tegrastats keeps running forever, so kill it after trying to read
-		// one line of output
-		err = cmd.Process.Signal(os.Kill)
-		if err != nil {
-			log.Errorf("unable to stop %s check: %s", checkName, err)
-		}
-	}()
 
-	// forward the standard error to the Agent logger
-	stderr, err := cmd.StderrPipe()
+	tegrastatsOutput, err := cmd.Output()
 	if err != nil {
-		return err
-	}
-	go func() {
-		in := bufio.NewScanner(stderr)
-		for in.Scan() {
-			log.Error(in.Text())
+		switch err := err.(type) {
+		case *exec.ExitError:
+			if len(tegrastatsOutput) <= 0 {
+				return fmt.Errorf("tegrastats did not produce any output: %s\n", err)
```

Comment:
CI is failing here, the fix should be
```suggestion
				return fmt.Errorf("tegrastats did not produce any output: %s", err)
```